### DISCUSSION TITLE
CS2 - Valve has decided to disallow Snap Tap functions and null-binds in Counter-Strike

### DISCUSSION THREAD

**Discussion by cafali** [cafali/SnapKey#68] [Announcements]
Valve's [latest update for CS2](https://www.counter-strike.net/newsentry/6500469346429600836?l=english) has officially disallowed the use of SnapKey and similar software/hardware solutions that manage keyboard input. These tools, including Snap Tap and others, are now disallowed on official Valve servers. 

If CS2 detects automated input from these sources, players will be kicked from the server (not banned) but can rejoin once the prohibited software or function is disabled.

Is this a positive change? 

SnapKey was designed to make this feature user-friendly and accessible to everyone.

When Snap Tap first gained popularity, it was locked behind a steep paywall, making it inaccessible to most players. 
Many gamers prefer simple, out-of-the-box solutions, rather than having to configure complex scripts or tools that aren't user-friendly for casual players.

Either allow it for everyone, or don't allow it at all.

With this update, Valve also removed the functionality of many other features, like jump-throw aliases and other custom binds. 

Feel free to use SnapKey in other games!

It is what it is...

**Comment by dillacorn** [comment]
Nothing you can do and I get that. Understandable not wanting to argue "SnapKey" validity to Valve.

They're banning a feature that could be implemented at the hardware level. It makes sense why they don't want the feature to be used.. ~but you can't truly stop something that is this easily accessible through many other undetectable methods.~

~I wouldn't be surprised if Valves anti-cheat will begin checking if you're using a Wooting or Razer Snap Tap capable keyboard and just prevent you from playing if those are detected on your system all together..~

**Reply by kam821** [reply]
> ALSO... what about less popular hall affect keyboards that are off everyones radar? If someone else implements this feature on a different keyboard that is NOT as popular as Wooting and Razer.. then what then? How can this feature truly be stopped...

Valve detects behavior, not specific hardware configuration, doesn't matter if SOCD is achieved via software or hardware.

**Reply by dillacorn** [reply]
## updated comment
[ahh here is the confirmation it is working as they say.. although not as intended currently..](https://youtu.be/rCDSpzH6-xw)

### this might affect hall affect keyboards negatively in CS!

### original comment
~I'd like a confirmation on this..~ I understand they're stating the anti-cheat is attempting to detect SOCD behavior ~but is it really?~

**Reply by minteeaa** [reply]
the detection is basically for frame-perfect inputs, which affects a ton of other keyboards in this field

unfortunately, Valve isn't too great at making proper solutions to issues that actually remain very prevalent in their games, and there's a good chance this ends up being one of those band-aid fixes they implement while they figure out how to manage community sentiment on it

**Comment by laazzee** [comment]
Thanks for your work ༼ つ ◕_◕ ༽つ
